elasticsearch_dsl/utils.py

```diff
@@ -19,33 +19,47 @@
 import collections.abc
 from copy import copy
 from typing import (
+    TYPE_CHECKING,
     Any,
     Callable,
     ClassVar,
     Dict,
     Generic,
     Iterable,
+    Iterator,
     List,
     Optional,
+    Tuple,
     Type,
     TypeVar,
     Union,
+    cast,
 )
 
 from typing_extensions import Self, TypeAlias
 
 from .exceptions import UnknownDslObject, ValidationException
 
-# Usefull types
+if TYPE_CHECKING:
+    from elastic_transport import ObjectApiResponse
+    from elasticsearch import AsyncElasticsearch, Elasticsearch
+
+    from .document_base import DocumentOptions
+    from .field import Field
+    from .index_base import IndexBase
+
+# Useful types
```

Comment:
nit: Thanks for fixing the typo, but that does not seem like a very useful comment!